code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -175,38 +175,73 @@
+		return FALSE
+	if(stat || paralysis || stunned || weakened ||  restrained())
+		src <<"<span class='warning'>You can't communicate while unable to move your hands to your head!</span>"
+		return FALSE
+	if(last_special > world.time)
+		src << "<span class='notice'>Your mind requires rest!</span>"
+		return FALSE
+
+	last_special = world.time + 40
+
+	visible_message("<span class='notice'>[src] touches their fingers to their temple.</span>")
 
 	var/list/targets = list()
-	var/target = null
+	for(var/mob/living/M in view(client.view, client.eye))
+		targets += M
+	var/mob/living/target = null
 	var/text = null
 
-	targets += getmobs() //Fill list, prompt user with list
 	target = input("Select a creature!", "Speak to creature", null, null) as null|anything in targets
 
-	if(!target) return
+	if(!target)
+		return
 
 	text = input("What would you like to say?", "Speak to creature", null, null)
 
 	text = sanitize(text)
 
-	if(!text) return
+	if(!text)
+		return
 
-	var/mob/M = targets[target]
+	if(istype(target, /mob/abstract/observer) || target.stat == DEAD)
+		to_chat(src, "<span class='warning'>Not even a [src.species.name] can speak to the dead.</span>")
+		return
 
-	if(istype(M, /mob/abstract/observer) || M.stat == DEAD)
-		src << "Not even a [src.species.name] can speak to the dead."
+	if (target.isSynthetic())
+		to_chat(src, "<span class='warning'>This can only be used on living organisms.</span>")
 		return
 
-	log_say("[key_name(src)] communed to [key_name(M)]: [text]",ckey=key_name(src))
 
-	M << "<span class='notice'>Like lead slabs crashing into the ocean, alien thoughts drop into your mind: [text]</span>"
-	if(istype(M,/mob/living/carbon/human))
-		var/mob/living/carbon/human/H = M
-		if(H.species.name == src.species.name)
-			return
-		H << "<span class='warning'>Your nose begins to bleed...</span>"
-		H.drip(1)
+	if (target.is_diona())
+		to_chat(src, "<span class='alium'>The creature's mind is not solid enough and slips through like sand.</span>")
+		return
+
+	if(!(target in view(client.view, client.eye)))
+		to_chat(src, "<span class='warning'>[target] is too far for your mind to grasp!</span>")
+		return
+
+	log_say("[key_name(src)] communed to [key_name(target)]: [text]",ckey=key_name(src))
+
+	var/mob/living/carbon/human/H = target
+	if (/mob/living/carbon/human/proc/commune in target.verbs)
+		target << "<span class='psychic'>You instinctively sense [src] sending their thoughts into your mind, hearing:</span> [text]"
+	else if(prob(25) && (target.mind && target.mind.assigned_role=="Chaplain"))
+		target<< "<span class='changeling'>You sense [src]'s thoughts enter your mind, whispering quietly:</span> [text]"
+	else
+		target << "<span class='alium'>You feel an ache behind your eyes as alien thoughts enter your mind:</span> [text]"
+		if(istype(target,/mob/living/carbon/human))
+			if (/mob/living/carbon/human/proc/commune in target.verbs)
+				return
+			H << "<span class='warning'>Your nose begins to bleed...</span>"
```

Comment:
Make sure the target species doesn't have the NO_BLOOD flag set.

Reply:
I would also put that message behind a chance based on the target species

Reply:
I'm pretty sure drip does this automatically, and anyone that does have NO_BLOOD is either immune or will literally never show up enough for it to matter.

Reply:
Insufficient.
There is a check for NO_BLOOD in the drip proc, so this would cause the nosebleed message and then no blood drips on the floor for species like Skeletons, Apparitions, Zombies and Golems which can occur during regular gameplay either being created by science or occuring in a gamemode.

Implement the check I outlined above.